docs/testplan/transceiver_onboarding/cmis_cdb_firmware_upgrade_testplan.md

```diff
@@ -0,0 +1,445 @@
+```
+
+##### 1.1.2 CMIS CDB Firmware Copy to DUT via sonic-mgmt infrastructure
+
+This section describes the automated process for copying firmware binaries to the DUT, ensuring only the required firmware versions are present for testing.
+
+**Firmware Selection Algorithm:**
+
+To ensure only the necessary firmware binaries are present for each transceiver:
+
+1. **Parse `transceiver_dut_info.csv`** to identify the transceiver present on the port identified by `port_under_test` test attribute. Generate the normalized vendor name and part number.
+2. **Parse `firmware_versions` test attribute** to get the next firmware version to download.
+3. **Parse `transceiver_firmware_info.csv`** to obtain the list of available firmware binaries, their versions, and associated vendor and part numbers. If the firmware version specified in step 2 exists for the normalized vendor name and part number from step 1:
+   - **Copy only the selected firmware binaries** to the target directory structure on the DUT.
+   - **Validate firmware binary integrity** using MD5 checksums after copying.
+   
+   Fail the test if the specified firmware version doesn't exist.
+
+**Cleanup:**
+
+- The firmware binary folder on the DUT (`/tmp/cmis_cdb_firmware/`) will be deleted after the test module run is complete to ensure a clean state for subsequent tests
+- Cleanup includes removing both the directory structure and any temporary files created during the process
+
+##### 1.1.3 CMIS CDB Firmware Upgrade Tests
+
+**Prerequisites:**
+
+1. **DOM polling must be disabled** to prevent race conditions between I2C transactions and the CDB mode for modules that cannot support CDB background mode.
+2. **Platform-specific processes:** On some platforms, `thermalctld` or similar user processes that perform I2C transactions with the module may need to be stopped during firmware operations.
+3. **Firmware requirements:**
+   - The firmware version specified  by `firmware_versions` test attribute must be available.
+   - All firmware versions must support the CDB protocol for proper testing.
+4. **Module capabilities:** The module must support dual banks for firmware upgrade operations.
+5. **Network connectivity:** The DUT must have network access to the firmware server specified in `cmis_cdb_firmware_base_url.csv` for downloading firmware binaries.
+
+| TC No. | Test | Steps | Expected Results |
+|------|------|------|------------------|
+| 1 | Firmware download validation | 1. Download the next firmware specified in `firmware_versions` using the sfputil CLI<br>2. Wait until CLI execution completes | 1. CLI execution should finish within `transceiver_download_timeout_minutes` minutes and return 0 <br>2. Active FW version should remain unchanged<br>3. Inactive FW version should reflect the downloaded firmware version<br> 4. No link flap should be seen<br>5. The kernel has no error messages in syslog<br>6. Critical process such as `xcvrd`, `syncd`  `orchagent` does not crash/restart. |
```

Comment:
"The kernel has no error messages in syslog" - this can be checked if monitor_kernel_errors is set.